crates/rpc/rpc-builder/src/eth.rs

```diff
@@ -112,3 +117,115 @@ impl EthConfig {
         self
     }
 }
+
+/// Context for building the `eth` namespace server.
+#[derive(Debug)]
+pub struct EthApiBuilderCtx<'a, Provider, Pool, EvmConfig, Network, Tasks, Events> {
+    /// Database handle.
+    pub provider: Provider,
+    /// Mempool handle.
+    pub pool: Pool,
+    /// Network handle.
+    pub network: Network,
+    /// EVM configuration.
+    pub evm_config: EvmConfig,
+    /// RPC config for `eth` namespace.
+    pub config: &'a EthConfig,
+    /// Runtime handle.
+    pub executor: Box<Tasks>,
+    /// Events handle.
+    pub events: Events,
+    /// RPC cache handle.
+    pub cache: EthStateCache,
+    /// Transaction forwarder used by `eth_sendRawTransaction` method.
+    pub raw_transaction_forwarder: Option<Arc<dyn RawTransactionForwarder>>,
+}
+
+impl<'a, Provider, Pool, EvmConfig, Network, Tasks, Events>
+    EthApiBuilderCtx<'a, Provider, Pool, EvmConfig, Network, Tasks, Events>
+{
+    /// Creates a new context for building the `eth` namespace server.
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        provider: Provider,
+        pool: Pool,
+        network: Network,
+        evm_config: EvmConfig,
+        config: &'a EthConfig,
+        executor: Box<Tasks>,
+        events: Events,
+        cache: EthStateCache,
+        raw_transaction_forwarder: Option<Arc<dyn RawTransactionForwarder>>,
+    ) -> Self {
+        Self {
+            provider,
+            pool,
+            network,
+            evm_config,
+            config,
+            executor,
+            events,
+            cache,
+            raw_transaction_forwarder,
+        }
+    }
```

Comment:
As all the fields of the `EthApiBuilderCtx` struct are public, are we sure that we need this kind of initializer with `too_many_arguments`? Maybe we can let the user initialize the structure himself, which amounts to the same thing because here we also have to repeat all the elements one by one, wdyt?

Reply:
this pr is one part in a big change to rpc design, so it is still prone to change. that's why it make sense to not spend time on making it perfect just yet.